### ISSUE TITLE
Unable to get a node demo working

### ISSUE THREAD

**Issue by gauntface** [highlight/highlight#7447]
**Describe the bug**
Following these steps to get highlight working for my node TypeScript project: https://www.highlight.io/docs/sdk/nodejs

I've run into a few issues trying to get this working.

1. The `highlight.run` and `@opentelemetry/*` modules were not installed after running `npm install --save @highlight-run/node`
1. `@highlight-run/node` seems to depend on `highlight.run` and that module references the type 	`HTMLCanvasElement` which isn't in the node environment.

    ```
    node_modules/highlight.run/dist/client/src/types/types.d.ts:261:25 - error TS2304: Cannot find name 'HTMLCanvasElement'.

    261     snapshot: (element: HTMLCanvasElement) => Promise<void>;
    ```

1. There is a module issue in highlight.run package (I honestly couldn't quite understand why this was an error, maybe the dist/package.json needs `"type": "module"` ?

    ```
    node_modules/highlight.run/dist/client/src/types/types.d.ts:1:115 - error TS2834: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.
    ```

**To Reproduce**

1. Create a typescript project with this tsconfig.json
    ```
    {
		"compilerOptions": {
			"module": "Node16",
			"target": "ES2022",
			"outDir": "out",
			"lib": [
				"ES2022"
			],
			"sourceMap": true,
			"rootDir": "src",
			"strict": true,   /* enable all strict type-checking options */
		}
    }
    ```
2. Follow https://www.highlight.io/docs/sdk/nodejs
3. Try to compile the project with `tsc -p ./`

**Expected behavior**
Compile successfully

**Comment by Vadman97** [comment]
hey @gauntface , wanted to clarify a few things:

1. `highlight.run` and `@opentelemetry/*` modules are not installed because they are not dependencies - they are bundled into `@highlight-run/node`'s entrypoints. Does that cause issues with your setup?
2. This seems to be a typescript 5 issue we've seen before. A workaround would be to enable `"skipLibCheck": true,` in the tsconfig. We're working on a fix in https://github.com/highlight/highlight/pull/7479
3. Your node runtime can pick up the HTML types if you change your `"lib"` definition to `"lib": ["dom", "ES2022"]`

**Comment by gauntface** [comment]
1. I just installed @highlight-run/node package and my build failed because it works highlight.run and @opentelemetry/* packages. Not sure where the fault was though
1. Noice!
1. Will have to check the lib definition

**Comment by Vadman97** [comment]
The typescript issue should be fixed with `@highlight-run/node@3.7.2`
Give it a try and let us know.
